Errors from mypy on CI

```
pytket/utils/results.py:154: error: Returning Any from function declared to return "ndarray[Any, Any]"  [no-any-return]
pytket/backends/backendresult.py:693: error: Dict entry 0 has incompatible type "str": "str"; expected "str": "list[Any]"  [dict-item]
pytket/backends/backendresult.py:693: error: Dict entry 1 has incompatible type "str": "str"; expected "str": "list[Any]"  [dict-item]
tests/transform_test.py:726: error: Argument 1 to "symbol_substitution" of "Circuit" has incompatible type "dict[Any, floating[Any]]"; expected "dict[Any, Any | float]"  [arg-type]
tests/transform_test.py:726: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tests/transform_test.py:726: note: Consider using "Mapping" instead, which is covariant in the value type
tests/transform_test.py:729: error: Argument 1 to "symbol_substitution" of "Circuit" has incompatible type "dict[Any, floating[Any]]"; expected "dict[Any, Any | float]"  [arg-type]
tests/transform_test.py:729: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tests/transform_test.py:729: note: Consider using "Mapping" instead, which is covariant in the value type
tests/transform_test.py:760: error: Argument 1 to "symbol_substitution" of "Circuit" has incompatible type "dict[Any, floating[Any]]"; expected "dict[Any, Any | float]"  [arg-type]
tests/transform_test.py:760: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tests/transform_test.py:760: note: Consider using "Mapping" instead, which is covariant in the value type
tests/transform_test.py:763: error: Argument 1 to "symbol_substitution" of "Circuit" has incompatible type "dict[Any, floating[Any]]"; expected "dict[Any, Any | float]"  [arg-type]
tests/transform_test.py:763: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tests/transform_test.py:763: note: Consider using "Mapping" instead, which is covariant in the value type
tests/transform_test.py:778: error: Argument 1 to "symbol_substitution" of "Circuit" has incompatible type "dict[Any, floating[Any]]"; expected "dict[Any, Any | float]"  [arg-type]
tests/transform_test.py:778: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tests/transform_test.py:778: note: Consider using "Mapping" instead, which is covariant in the value type
tests/transform_test.py:781: error: Argument 1 to "symbol_substitution" of "Circuit" has incompatible type "dict[Any, floating[Any]]"; expected "dict[Any, Any | float]"  [arg-type]
tests/transform_test.py:781: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tests/transform_test.py:781: note: Consider using "Mapping" instead, which is covariant in the value type
Found 9 errors in 3 files (checked 132 source files)
```

--- a/pytket/pytket/backends/backendresult.py
+++ b/pytket/pytket/backends/backendresult.py
@@ -690,7 +690,7 @@ def _check_permuted_sequence(first: Collection[Any], second: Collection[Any]) ->
 
 def _complex_ar_to_dict(ar: np.ndarray) -> dict[str, list]:
     """Dictionary of real, imaginary parts of complex array, each in list form."""
-    return {"real": ar.real.tolist(), "imag": ar.imag.tolist()}
+    return {"real": ar.real.tolist(), "imag": ar.imag.tolist()}  # type: ignore
 
 
 def _complex_ar_from_dict(dic: dict[str, list]) -> np.ndarray:

--- a/pytket/pytket/utils/results.py
+++ b/pytket/pytket/utils/results.py
@@ -151,7 +151,7 @@ def _compute_probs_from_state(state: np.ndarray, min_p: float = 1e-10) -> np.nda
     ignore = probs < min_p
     probs[ignore] = 0
     probs /= sum(probs)
-    return probs
+    return probs  # type: ignore
 
 
 def probs_from_state(

--- a/pytket/tests/transform_test.py
+++ b/pytket/tests/transform_test.py
@@ -723,10 +723,10 @@ def test_symbol_squash() -> None:
     for x in np.arange(0.0, 4.0, 0.4):
         smap = {a: x}
         c = circ.copy()
-        c.symbol_substitution(smap)
+        c.symbol_substitution(smap)  # type: ignore
         u = c.get_unitary()
         c1 = circ1.copy()
-        c1.symbol_substitution(smap)
+        c1.symbol_substitution(smap)  # type: ignore
         u1 = c1.get_unitary()
         # PauliSquash does not preserve global phase.
         v = u @ u1.conjugate().transpose()
@@ -757,10 +757,10 @@ def test_symbol_pauli_squash_1() -> None:
     for x in np.arange(0.0, 4.0, 0.4):
         smap = {Symbol("a"): x}
         c = circ.copy()
-        c.symbol_substitution(smap)
+        c.symbol_substitution(smap)  # type: ignore
         u = c.get_unitary()
         c1 = circ1.copy()
-        c1.symbol_substitution(smap)
+        c1.symbol_substitution(smap)  # type: ignore
         u1 = c1.get_unitary()
         # PauliSquash does not preserve global phase.
         v = u @ u1.conjugate().transpose()
@@ -775,10 +775,10 @@ def test_symbol_pauli_squash_2() -> None:
     for x in np.arange(0.0, 4.0, 0.4):
         smap = {Symbol("a"): x}
         c = circ.copy()
-        c.symbol_substitution(smap)
+        c.symbol_substitution(smap)  # type: ignore
         u = c.get_unitary()
         c1 = circ1.copy()
-        c1.symbol_substitution(smap)
+        c1.symbol_substitution(smap)  # type: ignore
         u1 = c1.get_unitary()
         # PauliSquash does not preserve global phase.
         v = u @ u1.conjugate().transpose()

